### PR TITLE
#166790011 Share Articles Across Different Channels

### PR DESCRIPTION
**Pull request by IsaiahRn**
#### What does this PR do?

- Users should be able to share articles across different channels

#### Description of Task to be completed

- Add endpoint for sharing articles to `Facebook`, `Twitter` or `Mail`

- Check existing article's slug

- Add tests for the endpoint

- Document the endpoint

#### How should this be manually tested?

1. Run the following commands in the terminal.
```
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-share-articles-166790011
npm install
npm start
```

2. Test endpoints, in Postman:

- Share on Twitter via `http://localhost:3000/api/articles/:slug/share/twitter`
- Share on Facebook via `http://localhost:3000/api/articles/:slug/share/facebook`
- Share on Mail via `http://localhost:3000/api/articles/:slug/share/mail`

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?
[#166790011](https://www.pivotaltracker.com/story/show/166790011)
#### Screenshots (if appropriate)

- Shared on Twitter
Screen Shot 2019-07-21 at 18 51 53

- Shared on Facebook
Screen Shot 2019-07-21 at 18 50 38

- Shared on Mail
Screen Shot 2019-07-21 at 18 58 46

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by Hervera — DISMISSED** [review]
Ready to be merged

File: controllers/article.js [hunk 1]
```diff
@@ -187,6 +188,51 @@ class Article {
...
     }
   }
+
+  /**
+   *
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {object} response
+   */
+  static async share(req, res) {
```
Comment (kodek-sleuth) [inline controllers/article.js]: Please rename this method to 'shareArticle'

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

File: tests/article.test.js [hunk 2]
```diff
@@ -247,6 +248,82 @@ describe('CRUD articles routes', () => {
...
+      .then((res) => {
+        res.should.have.status(200);
+        authToken = res.body.user.token; // get the token
+        done();
+      })
+      .catch(err => done(err));
+  });
+
+  // @create article
+  it('should create an article', (done) => {
```
Comment (kodek-sleuth) [inline tests/article.test.js]: Please use a seeded article

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

**Review by niyoceles — APPROVED** [review]
good job

**Review by kodek-sleuth — APPROVED** [review]

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Pull request merged**